commcare_export/builtin_locations_query.py

```diff
@@ -0,0 +1,231 @@
+
+TABLE_NAME = "commcare_locations"
+
+# A JSON format MiniLinq query for internal CommCare location table.
+# It reads every field produced by the /location/ API endpoint and
+# writes the data to a table named "commcare_locations" in a database.
+jvalue = {
```

Comment:
one issue with having this as the full query is that if there are changes to how we compile queries in the future this may not continue to work without applying the appropriate changes. One way to 'future proof' this would be to store this as a set of `SheetParts` and then use `compile_queries` to convert that into the final query. I think that would also make this much less verbose:

```python

class Column:
    def __init__(name, source, map_function=None, *extra_args):
        ...

    @property
    def output_field(self):
        if not self.map_function:
            return Reference(self.source)
        else:
            return Apply(Reference(self.map_function), Reference(self.source), *self.extra_args)

columns = [
    Column('id', 'id'),
    Column('created_at', 'created_at', 'str2date'),
    ...
    Column('location_type_administrative', 'location_type', 'get_location_info, 'administrative')
    ...
]
source = Apply(Reference("api_data"), Literal(data_source), Reference('checkpoint_manager'))
part = SheetParts(TABLE_NAME, [c.name for c in columns], source, List([c.output_field for c in columns], None)
query = compile_queries([part], None, False)

Reply:
I like it. Done.